Documentation/ContentObjects/Extbaseplugin/_CodeSnippets/_MyController.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MyVendor\MyExtension\Controller;
+
+use Psr\Http\Message\ResponseInterface;
+use TYPO3\CMS\Extbase\Mvc\Controller\ActionController;
+use TYPO3\CMS\Frontend\ContentObject\ContentObjectRenderer;
+
+class MyController extends ActionController
+{
+    public function listAction(): ResponseInterface
+    {
+        /** @var ContentObjectRenderer $contentObject */
+        $contentObject = $this->request->getAttribute('currentContentObject');
+        $dataFromTypoScript = $contentObject->getCurrentVal();
```

Comment:
`getCurrentVal()` returns 98% string or int and represents in most cases just a single value of `COR->data` property. Are you sure the example is correct? Maybe you mean `$contentObject->data` instead.

Reply:
I will check again after my vacation

Reply:
I tested this: You are correct you have to use data if dealing with array, getCurrentVal() if dealing with strings. Adjusted the code accordingly